app/controllers/concerns/hal_actions/actions.rb

```diff
@@ -0,0 +1,104 @@
+module HalActions::Actions
+  def index
+    respond_with index_collection, index_options
+  end
+
+  def show
+    respond_with show_resource, show_options
+  end
+
+  def create
+    create_resource.tap do |resource|
+      consume_with_content_type! resource
+      authorize resource
+      resource.save!
+      respond_with root_resource(resource), show_options
+    end
+  end
+
+  def update
+    update_resource.tap do |resource|
+      consume_with_content_type! resource
+      authorize resource
+      resource.save!
+      respond_with root_resource(resource), show_options
+    end
+  end
+
+  def destroy
+    destroy_resource.tap do |resource|
+      authorize resource
+      resource.destroy
+      redirect_to destroy_redirect
```

Comment:
I would not expect a redirect response for a successful API call to destroy, more likely a 20(0|2|4) response, right?